sec_certs/dataset.py

```diff
@@ -1108,32 +1134,48 @@ def color_check(current_key):
                            (self.certs[current_key].web_scan.module_name
                             if self.certs[current_key].web_scan.module_name else ''))
 
+        def get_processed_list():
```

Comment:
Isn't it a code smell? Inner function to touch the global arguments implicitly? Anyway, these 9 lines could be rewritten as:

```python
def get_processed_list(connection_list: str, key:):
    attr = {'pdf': 'pdf_scan', 'web': 'web_scan', 'processed': 'processed'}[connection_list]
    return getattr(self.certs[key], attr).connections
```